stable/openstack-telemetry/README.md

```diff
@@ -1,9 +1,308 @@
+overlay bundles.
+
+See the Juju documentation on [network spaces][juju-spaces] and [overlay
+bundles][juju-overlays] for background information.
+
+## MAAS cloud and Juju controller
+
+Ensure that the MAAS cluster has been added to Juju as a cloud and that a Juju
+controller has been created for that cloud. See the Juju documentation for
+guidance: [Using MAAS with Juju][juju-and-maas].
+
+Assuming the controller is called 'maas-controller' and you want to use the
+empty model 'default', change to that context:
+
+    juju switch maas-controller:default
+
+## Deploy the cloud
+
+First move into the bundle directory:
+
+    cd ~/openstack-telemetry
+
+To install OpenStack use this command if you're using the spaces overlay:
+
+    juju deploy ./bundle.yaml --overlay ./openstack-telemetry-spaces-overlay.yaml
+
+Otherwise, simply do:
+
+    juju deploy ./bundle.yaml
+
+## Install the OpenStack clients
+
+You'll need the OpenStack clients in order to manage your cloud from the
+command line. Install them now:
+
+    sudo snap install openstackclients
+
+## Unseal vault
+
+This release uses vault to secure keystone - vault needs to be unsealed before the configuration
+can be finalised and the cloud used. See the following documentation for unseal steps:
+
+[Unsealing vault][vault-appendix]
+
+## Access the cloud
+
+Confirm that you can access your cloud from the command line:
+
+    source ~/openstack-telemetry/openrc
+    openstack service list
+
+You should get a listing of all registered cloud services.
+
+## Import an image
+
+You'll need to import a boot image in order to create instances. Here we
+import a Bionic amd64 image and call it 'bionic':
+
+    curl http://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img | \
+        openstack image create --public --container-format=bare --disk-format=qcow2 \
+        bionic
+
+Images for other Ubuntu releases and architectures can be obtained in a similar
+way.
+
+For the ARM 64-bit (arm64) architecture you will need to configure the image to
+boot in UEFI mode:
+
+    curl http://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-arm64.img | \
+        openstack image create --public --container-format=bare --disk-format=qcow2 \
+        --property hw_firmware_type=uefi bionic
+
+## Configure networking
+
+Networking will be configured with the aid of two scripts supplied by
+the bundle.
+
+> **Note**: These scripts are written for Python 2 and may not work on a modern
+  system without first installing the ``python-keystoneclient`` and
+  ``python-neutronclient`` deb packages.
+
+> **NOTE**: The network type required for OVN is geneve - the default for the
+  neutron-ext-net scripts is currently gre.
+
+For the "external" network a shared router ('provider-router') will be used by
+all tenants for public access to instances. The syntax is:
+
+    ./neutron-ext-net-ksv3 --network-type geneve \
```

Comment:
Same comment as above, I believe `flat` or `vlan` type should be used here instead of geneve.